prtpy/partitioning/complete_greedy.py

```diff
@@ -134,108 +199,129 @@ def anytime(
+                    else:
+                        fast_lower_bound = 0
+                        avg = sum(items) / numbins
+                        for i in range (numbins):
+                            fast_lower_bound = fast_lower_bound + max(current_sums[i]-avg,0)
                 else:
                     fast_lower_bound = -np.inf
                 if fast_lower_bound >= best_objective_value:
                     times_fast_lower_bound_activated += 1
                     continue
 
             new_bins = binner.add_item_to_bin(binner.copy_bins(current_bins), next_item, bin_index)
-            binner.sort_by_ascending_sum(new_bins)
+            if not relative_value:
+                binner.sort_by_ascending_sum(new_bins)
             new_sums = tuple(binner.sums(new_bins))
 
             # Lower-bound heuristic. 
             if use_lower_bound:
-                lower_bound = objective.lower_bound(new_sums, sum_of_remaining_items, are_sums_in_ascending_order=True)
+                lower_bound = objective.lower_bound(new_sums, sum_of_remaining_items, are_sums_in_ascending_order=False)
                 if lower_bound >= best_objective_value:
                     logger.debug("    Lower bound %f too large", lower_bound)
                     times_lower_bound_activated += 1
                     continue
-            if use_set_of_seen_states: 
+            if use_set_of_seen_states:
                 if new_sums in seen_states:
                     logger.debug("    State %s already seen", new_sums)
                     times_seen_state_skipped += 1
                     continue
-                seen_states.add(new_sums)   # should be after if use_lower_bound
+                seen_states.add(new_sums)  # should be after if use_lower_bound
 
             new_vertex = (new_bins, depth + 1)
             stack.append(new_vertex)
             intermediate_partitions_checked += 1
 
-    logger.info("Checked %d out of %d complete partitions, and %d intermediate partitions.", complete_partitions_checked, numbins**numitems, intermediate_partitions_checked)
-    logger.info("  Heuristics: fast lower bound = %d, lower bound = %d, seen state = %d, heuristic 3 = %d.", times_fast_lower_bound_activated, times_lower_bound_activated, times_seen_state_skipped, times_heuristic_3_activated)
+    logger.info("Checked %d out of %d complete partitions, and %d intermediate partitions.",
+                complete_partitions_checked, numbins ** numitems, intermediate_partitions_checked)
+    logger.info("  Heuristics: fast lower bound = %d, lower bound = %d, seen state = %d, heuristic 3 = %d.",
+                times_fast_lower_bound_activated, times_lower_bound_activated, times_seen_state_skipped,
+                times_heuristic_3_activated)
 
+
+    if (relative_value):
+        for i in range(numbins):
+            binner.remove_item_from_bin(best_bins, i, 0)
+
+    for i in range(numbins):
+        binner.sums(best_bins)[i] = math.floor(binner.sums(best_bins)[i])
```

Comment:
Why are the sums rounded to integers?

Reply:
The sums were supposed to be integers the whole time but with relative values - I needed to add a double value into each one of the bins in the beggining of the algorithm. This gave me issues, for example some values were something.00000000000001 so I just turned them into integers after removing the double value.